### PR TITLE
[FEATURE] Add if() expression support to ExpressionBuilder

### PR DESCRIPTION
**Pull request by brotkrueml**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/977
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/979
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst [hunk 1]
```diff
@@ -266,6 +266,56 @@ For PostgreSQL the :sql:`"value"::INTEGER` cast notation is used.
...
+..  _database-expression-builder-if:
+
+:php:`ExpressionBuilder::if()`
+------------------------------
+
+..  versionadded:: 13.3
+
+..  include:: _ExpressionBuilderIf.rst.txt
+
+This method enables to phrase "if-then-else" expressions. Those are
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:279]:
```suggestion
This method is used for "if-then-else" expressions. These are
```

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst [hunk 2]
```diff
@@ -266,6 +266,56 @@ For PostgreSQL the :sql:`"value"::INTEGER` cast notation is used.
...
+translated into :sql:`IF` or :sql:`CASE` statements depending on the
+used database engine.
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:281]:
```suggestion
database engine in use.
```

File: Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst [hunk 3]
```diff
@@ -266,6 +266,56 @@ For PostgreSQL the :sql:`"value"::INTEGER` cast notation is used.
...
+
+..  warning::
+    No automatic quoting or escaping is done for the condition and true/false
+    part. Extension authors need to ensure proper quoting for each part or use
+    API calls doing the quoting, for example the
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst:286]:
```suggestion
    API calls for doing the quoting, for example the
```

**Review by linawolf — COMMENTED** [review]
This is Sarah from Lina's account :-)

**Review by garvinhicking — APPROVED** [review]
Nice, thanks! Giving a pre+1, I think Sarahlinas changes are good,
too!

**Review by garvinhicking — APPROVED** [review]

**Review by linawolf — APPROVED** [review]

**Comment by linawolf** [comment]
Welcome back on board!

**Pull request merged**